Fix saving translations when only translations are modified.

markstory opened the pull request:
When we have bundled translations but no changed fields, we need to mark the original fields dirty so we can save the root entity and the attached translations.

Refs #9610

andrej-griniuk committed:
add translate behaviour test refs #9610

andrej-griniuk committed:
test update

markstory committed:
Fix new entities with no fields, and bundled translations not saving.

When we have bundled translations but no changed fields, we need to mark
the original fields dirty so we can save the root entity and the
attached translations.

Refs #9610

markstory committed:
Add additional test covering update with only translations.

Refs #9610

thinkingmedia commented on src/ORM/Behavior/TranslateBehavior.php:
```diff
@@ -271,22 +271,35 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
+            foreach ($this->_config['fields'] as $field) {
+                $entity->dirty($field, true);
+            }
+            return;
```
🙄

markstory committed:
Fix PHPCS.

lorenzo commented on src/ORM/Behavior/TranslateBehavior.php:
```diff
@@ -271,22 +271,36 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
         $fields = array_keys($values);
+        $noFields = empty($fields);
+
+        $primaryKey = (array)$this->_table->primaryKey();
```
Why move these lines up here?

markstory replied:
I can move them back down.

markstory committed:
Move variables closer to where they are used.

markstory commented:
I'll look at the postgres builds tonight.

markstory committed:
Only dirty fields when the entity is new.

Postgres did not like the previous changes as they caused additional
update queries and ids to mismatch.

lorenzo commented:
Great job!
